docs/DeveloperGuide.md

```diff
@@ -1,37 +1,209 @@
+
+![Find_Sequence_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/FindSequenceDiagram.png)
+
+### **State Switching Feature**
+
+The state switching feature allows the system to change its mode of operation between `MAIN_STATE` (for patient management) and `TASK_STATE` (for task management). This allows the software to handle commands differently based on the current state.
+
+1. **User Input**: The user enters a command pertaining either to patients (in `MAIN_STATE`) or tasks (in `TASK_STATE`).
+2. **Command Parsing**: The `Parser` checks the current state of the software by interacting with the `State` class. Depending on the state, it interprets  the command differently.
+3. **State Checking**: The `State` class tracks the current state of the application. If the system is in `MAIN_STATE`, the `Parser` creates patient-related commands (e.g., `AddPatientCommand`). If it's in `TASK_STATE`, task-related commands (e.g., `AddTaskCommand`)
+4. **Command Execution**: The appropriate command is then executed. For example, in `MAIN_STATE`, it adds a new patient, while in `TASK_STATE`, it adds a new task to a patient's task list.
+5. **State Transitions**: Commands like `SelectPatientCommand` and `BackCommand`, trigger state transitions.
+
+#### Sequence Diagram
+
+The following sequence diagram illustrates how the state-switching mechanism works:
+
+![State-Switching Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/StateSwitchingSequenceDiagram.png)
+
+- **State Checking**: When user inputs a command, the `Parser` queries the `State` object to check whether the system is in `MAIN_STATE` or `TASK_STATE`.
+- **State Transitions**: When a `SelectPatientCommand` is executed, the system transitions from `MAIN_STATE` to `TASK_STATE`. Vice versa for `BackCommand`.
+- **Command Execution**: Commands are executed based on the current state. For example, in `MAIN_STATE`, patient-related commands are processed, while in `TASK_STATE`, task-related commands are handled.
 
 ## Product scope
+
 ### Target user profile
 
-{Describe the target user profile}
+- Nurses who need to manage their tasks for each patient with tagging.
+- Nurses who need to keep track of their tasks for each patient.
+- Is reasonably comfortable with the command line interface.
+- Is familiar with the concept of tasks and patients.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+- MediTask is a handy tool for nurses to coordinate their tasks according to patients.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ... | So that I can ... |
```

Comment:
Would be good if details about what type of nurses could benefit specifically, but overall very comprehensive and clear with the "so that i can..." section!